spec/mongo/error/bulk_write_error_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'lite_spec_helper'
+
+describe Mongo::Error::BulkWriteError do
+  let(:result) do
+    {
+      'writeErrors' => [
+        { 'code' => 1, 'errmsg' => 'message1' },
+        { 'code' => 2, 'errmsg' => 'message2' },
+      ]
+    }
+  end
+
+  let(:error) { described_class.new(result) }
+
+  let(:messages) do
+    'message1 (1), message2 (2)'
+  end
+
+  let(:notes_tail) do
+    ' (note1, note2)'
+  end
+
+  before do
+    error.add_note('note1')
+    error.add_note('note2')
+  end
+
+  describe '#result' do
+    it 'returns the result' do
+      expect(error.result).to eq(result)
+    end
+  end
+
+  describe '#labels' do
+    it 'returns an empty array' do
+      expect(error.labels).to eq([])
+    end
+  end
+
+  describe '#message' do
+    it 'returns the formatted message' do
+      expect(error.message).to eq("#{described_class}: #{messages}#{notes_tail}")
+    end
+  end
+
+  describe '#to_s' do
+    it 'returns the error represented as a string' do
+      expect(error.to_s).to eq("#{described_class}: #{messages}#{notes_tail}")
```

Comment:
I tried looking at the error object by itself:

<pre>
[43, 52] in /home/w/apps/ruby-driver/spec/mongo/error/bulk_write_error_spec.rb
   43:     end
   44:   end
   45: 
   46:   describe '#to_s' do
   47:     it 'returns the error represented as a string' do
   48:     byebug
=> 49:       expect(error.to_s).to eq("#{described_class}: #{messages}#{notes_tail}")
   50:     end
   51:   end
   52: end
(byebug) error
#&lt;Mongo::Error::BulkWriteError: Mongo::Error::BulkWriteError: message1 (1), message2 (2) (note1, note2) (note1, note2)>
(byebug) error.to_s
"Mongo::Error::BulkWriteError: message1 (1), message2 (2) (note1, note2)"
(byebug) 

</pre>